server/modules/turf/turf.controller.ts

```
@@ -146,99 +176,133 @@ export default class TurfController {
+
+      // Find first error, if any
+      const error = validationResults.find(result => result.error);
+      if (error) {
+        return next(error.error);
+      }
 
-    return updateData;
+      // Merge all valid results
+      const parsedData = validationResults.reduce(
+        (acc, result) => ({
+          ...acc,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)